Sources/TokamakLint/Protocols/Rule.swift

```diff
@@ -0,0 +1,23 @@
+//
+//  Rule.swift
+//  TokamakCLI
+//
+//  Created by Matvii Hodovaniuk on 4/9/19.
+//
+
+import Foundation
+
+public protocol Rule {
+  static var description: RuleDescription { get }
+
+  init()
+
+  static func validate(path: String) throws -> [StyleViolation]
+  static func validate(visitor: TokenVisitor) -> [StyleViolation]
```

Comment:
is there any benefit in requiring rules to implement 2 functions at once, while one can be inferred from the other?